zeppelin-server/src/main/java/org/apache/zeppelin/server/JsonExclusionStrategy.java

```diff
@@ -20,20 +20,35 @@
 import com.google.gson.ExclusionStrategy;
 import com.google.gson.FieldAttributes;
 import org.apache.zeppelin.interpreter.InterpreterOption;
+import org.apache.zeppelin.notebook.Note;
+import org.apache.zeppelin.notebook.Paragraph;
+import org.apache.zeppelin.display.AngularObject;
+
+import java.util.List;
 
 /**
  * Created by eranw on 8/30/15.
- * Omit InterpreterOption from serialization
+ * Omit classes from serialization
  */
 public class JsonExclusionStrategy implements ExclusionStrategy {
 
   public boolean shouldSkipClass(Class<?> arg0) {
-    //exclude only InterpreterOption
-    return InterpreterOption.class.equals(arg0);
+
+    return (InterpreterOption.class.equals(arg0) ||
+        Paragraph.class.equals(arg0) ||
+        AngularObject.class.equals(arg0));
   }
 
   public boolean shouldSkipField(FieldAttributes f) {
 
-    return false;
+    Boolean shouldSkip = false;
+    // Exclude Note angularObjects
+    if (f.getDeclaringClass().equals(Note.class) && f.getName().equals("angularObjects")) {
+      shouldSkip = true;
+      // Exclude Note paragraphs
+    } else if (f.getDeclaringClass().equals(Note.class) && f.getName().equals("paragraphs")) {
```

Comment:
I don't understand the difference but don't mind switch the order.
This will only apply to paragraph properties within the Note.class so other paragraphs fields will not be effected.